tests/end-to-end/backup-restore-test/utils/config/config.go

```diff
@@ -0,0 +1,24 @@
+package config
+
+import (
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	"os"
+)
+
+func NewRestClientConfig() (*rest.Config, error) {
+	kubeconfigPath := os.Getenv("KUBECONFIG")
+
+	var config *rest.Config
+	var err error
+	if kubeconfigPath != "" {
+		config, err = clientcmd.BuildConfigFromFlags("", kubeconfigPath)
+	} else {
+		config, err = rest.InClusterConfig()
+	}
```

Comment:
thats a bit redundant: https://github.com/kubernetes/client-go/blob/48376054912de15b6386e4310192c4e8aab98403/tools/clientcmd/client_config.go#L547

This entire functionality is already builtin to BuildConfigFromFlags